extensions/usAnnotation/src/panels/USAnnotationPanel.tsx

```diff
@@ -0,0 +1,423 @@
+                  className={`border-b border-blue-900 ${
+                    item.frame === 5 ? 'bg-cyan-800 bg-opacity-30' : ''
+                  }`}
+                  onClick={() => handleRowClick(item)}
+                  style={{ cursor: 'pointer' }}
+                >
+                  <td className="py-3 px-3">{item.index}</td>
+                  <td className="py-3 px-3">{item.frame}</td>
+                  <td className="py-3 px-3 text-center">{item.pleura}</td>
+                  <td className="py-3 px-3 text-center">{item.bLine}</td>
+                  <td className="py-3 px-3 text-right">
+                    {item.frame === 5 && (
+                      <div className="flex items-center justify-end">
+                        <Button variant="ghost" className="p-0 text-blue-300">
+                          <Icons.EyeVisible />
+                        </Button>
+                        <Button variant="ghost" className="ml-2 p-0 text-blue-300">
+                          <Icons.More />
+                        </Button>
+                      </div>
+                    )}
+                  </td>
+                </tr>
+              ))}
+            </tbody>
+          </table>
+        </div>
+      </PanelSection.Content>
+    </ScrollArea>
+  );
+
+  /**
+   * Callback function that is called when an annotation is modified
+   * Updates the annotatedFrames state with the latest annotation data
+   */
+  const annotationModified = React.useCallback(
+    event => {
+      if (event.detail.annotation.metadata.toolName === UltrasoundAnnotationTool.toolName) {
+        const activeViewportId = viewportGridService.getActiveViewportId();
+        const viewport = cornerstoneViewportService.getCornerstoneViewport(activeViewportId);
+        // copying to avoid mutating the original array
+        const imageIdsMonitored = [...imageIdsToObserve];
+        const imageIdFilter = (imageId: string) => {
+          if (imageIdsMonitored.length === 0) {
+            return true;
+          }
+          return imageIdsMonitored.includes(imageId);
+        };
+        const mapping = UltrasoundAnnotationTool.countAnnotations(viewport.element, imageIdFilter);
+        const keys = Array.from(mapping.keys());
+        const updatedFrames = keys.map((key, index) => {
+          const { pleura, bLine, frame } = mapping.get(key) || { pleura: 0, bLine: 0, frame: 0 };
+          return {
+            imageId: key,
+            index: index + 1,
+            frame,
+            pleura,
+            bLine,
+          };
+        });
+        setAnnotatedFrames(updatedFrames);
+      }
+    },
+    [viewportGridService, cornerstoneViewportService, imageIdsToObserve]
+  );
+
+  useEffect(() => {
+    eventTarget.addEventListener(csToolsEnums.Events.ANNOTATION_MODIFIED, annotationModified);
+
+    return () => {
+      eventTarget.removeEventListener(csToolsEnums.Events.ANNOTATION_MODIFIED, annotationModified);
+    };
+  }, [annotationModified]);
```

Comment:
I thought this panel would use `useMeasurements` with a filter for the ultrasound annotation tool type. Instead of adding an event listener to the event target for this, take a look at how to use `useMeasurements`, or how to modify it, so we can remove this subscription.